File: lib/models/download_configuration.dart
class DownloadConfiguration {
  const DownloadConfiguration({
    this.title = '',
    required this.url,
  });

  final String title;
  final String url;

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map['title'] = title;
    map['url'] = url;
    return map;
  }

  @override
  String toString() => 'DownloadConfiguration{title: $title, url: $url}';
}

File: ios/Classes/Player/Models/DownloadConfiguration.swift
//
//  DownloadConfiguration.swift
//  video_player
//
//  Created by Sunnatillo Shavkatov on 23/06/25.
//

import Foundation

struct DownloadConfiguration {
    var url: String
    var title: String
    
    init(url: String, title: String) {
        self.url = url
        self.title = title
    }
    
    static func fromMap(map: [String: Any]) -> DownloadConfiguration? {
        guard let url = map["url"] as? String,
              let title = map["title"] as? String else {
            return nil
        }
        return DownloadConfiguration(url: url, title: title)
    }
}
